app/src/main/java/com/concordia/mcga/lib/BuildingBottomSheetInfo.java

```diff
@@ -0,0 +1,867 @@
+                !mIgnoreEvents && mState != STATE_DRAGGING &&
+                !parent.isPointInChildBounds(scroll, (int) event.getX(), (int) event.getY()) &&
+                Math.abs(mInitialY - event.getY()) > mViewDragHelper.getTouchSlop();
+        return ret;
+    }
+
+    @Override
+    public boolean onTouchEvent(CoordinatorLayout parent, V child, MotionEvent event ) {
+        if ( ! child.isShown() ) {
+            return false;
+        }
+
+        int action = MotionEventCompat.getActionMasked( event );
+        if ( mState == STATE_DRAGGING  &&  action == MotionEvent.ACTION_DOWN ) {
+            return true;
+        }
+
+        mViewDragHelper.processTouchEvent( event );
+
+        if ( action == MotionEvent.ACTION_DOWN ) {
+            reset();
+        }
+
+        // The ViewDragHelper tries to capture only the top-most View. We have to explicitly tell it
+        // to capture the bottom sheet in case it is not captured and the touch slop is passed.
+        if ( action == MotionEvent.ACTION_MOVE  &&  ! mIgnoreEvents ) {
+            if ( Math.abs(mInitialY - event.getY()) > mViewDragHelper.getTouchSlop() ) {
+                mViewDragHelper.captureChildView( child, event.getPointerId(event.getActionIndex()) );
+            }
+        }
+        return ! mIgnoreEvents;
+    }
+
+    @Override
+    public boolean onStartNestedScroll(CoordinatorLayout coordinatorLayout, V child, View directTargetChild, View target, int nestedScrollAxes ) {
+        mNestedScrolled = false;
+        return ( nestedScrollAxes & ViewCompat.SCROLL_AXIS_VERTICAL ) != 0;
+    }
+
+    private ScrollVelocityTracker mScrollVelocityTracker = new ScrollVelocityTracker();
+    private class ScrollVelocityTracker {
+        private long  mPreviousScrollTime = 0;
+        private float mScrollVelocity     = 0;
+
+        public void recordScroll( int dy ) {
+            long now = System.currentTimeMillis();
+
+            if ( mPreviousScrollTime != 0 ) {
+                long elapsed = now - mPreviousScrollTime;
+                mScrollVelocity = (float) dy / elapsed * 1000; // pixels per sec
+            }
+
+            mPreviousScrollTime = now;
+        }
+
+        public void clear() {
+            mPreviousScrollTime = 0;
+            mScrollVelocity = 0;
+        }
+
+        public float getScrollVelocity() {
+            return mScrollVelocity;
+        }
+    }
+
+    @Override
+    public void onNestedPreScroll(CoordinatorLayout coordinatorLayout, V child, View target, int dx, int dy, int[] consumed ) {
+        View scrollingChild = mNestedScrollingChildRef.get();
+        if ( target != scrollingChild ) {
+            return;
+        }
+
+        mScrollVelocityTracker.recordScroll( dy );
+
+        int currentTop = child.getTop();
+        int newTop     = currentTop - dy;
+
+        // Force stop at the anchor - do not go from collapsed to expanded in one scroll
+        /*
```

Comment:
All the code history can be found on github if needed in the futur. Please don't leave dead code